test/common/http/http2/conn_pool_test.cc

```diff
@@ -72,44 +73,82 @@ class Http2ConnPoolImplTest : public testing::Test {
     EXPECT_EQ("", TestUtility::nonZeroedGauges(cluster_->stats_store_.gauges()));
   }
 
-  TestCodecClient& createTestClient() {
-    test_clients_.emplace_back();
-    TestCodecClient& test_client = test_clients_.back();
-    test_client.connection_ = new NiceMock<Network::MockClientConnection>();
-    test_client.codec_ = new NiceMock<Http::MockClientConnection>();
-    test_client.connect_timer_ = new NiceMock<Event::MockTimer>(&dispatcher_);
-    test_client.client_dispatcher_ = api_->allocateDispatcher("test_thread");
-    EXPECT_CALL(*test_client.connect_timer_, enableTimer(_, _));
+  void createTestClients(int num_clients) {
```

Comment:
Should we have similar tests for H1 connection pools?

Reply:
It's testing the same underlying code - and by doing tests with both multiplexing eq 1 and gt 1 I think we unit test both use cases appropriately (and the integration test makes sure it actually works).  Ideally we'd have a unit test for the base class but that wasn't implemented when the connection pool went shared.

Reply:
Thanks for the explanation.  Testing base class behavior is sometimes challenging, sounds like this is as good as we can do right now.